src/models/helpers/admin.rs

```diff
@@ -0,0 +1,90 @@
+use std::{collections::HashSet, num::ParseIntError, str::FromStr};
+
+use once_cell::sync::Lazy;
+
+use crate::util::errors::{forbidden, AppResult};
+
+static AUTHORIZED_ADMIN_USER_IDS: Lazy<HashSet<i32>> =
+    Lazy::new(|| parse_authorized_admin_users(dotenvy::var("GH_ADMIN_USER_IDS")));
+
+// The defaults correspond to the current crates.io team, which as at the time of writing, is the
+// GitHub user names carols10cents, jtgeibel, Turbo87, JohnTitor, LawnGnome, and mdtro.
+//
+// FIXME: this needs to be removed once we can detect the admins from the Rust teams API.
+const DEFAULT_ADMIN_USER_IDS: [i32; 6] = [193874, 22186, 141300, 25030997, 229984, 20070360];
+
+fn parse_authorized_admin_users(maybe_users: dotenvy::Result<String>) -> HashSet<i32> {
+    match maybe_users {
+        Ok(users) => users
+            .split(|c: char| !(c.is_ascii_digit()))
+            .filter(|uid| !uid.is_empty())
+            .map(i32::from_str)
+            .collect::<Result<HashSet<i32>, ParseIntError>>()
+            .expect("parsing admin users from $GH_ADMIN_USER_IDS"),
+        Err(_err) => DEFAULT_ADMIN_USER_IDS.into_iter().collect(),
+    }
+}
```

Comment:
now that I think about it some more, we might want to put this into the `config.rs` file, next to all the other env var server config code. the `config::Server` struct already has a dedicated implementation for our test suite, so the `#[cfg(test)]` below on the `check_user_id()` fn wouldn't be necessary anymore.

one downside of this is that we need access to that config struct when checking if someone is an admin user, but in `AuthCheck::check()` we have access to the `Request` struct, which iirc should give use access to the state extension and thus access to the config struct too.

Reply:
Will investigate, either today if I have some free time at the conference or next week when I get back from my long weekend Ucluelet.